core/src/Threads/Kokkos_Threads_Team.hpp

```diff
@@ -188,8 +188,6 @@ class ThreadsExecTeamMember {
         using type =
             typename if_c<sizeof(Type) < TEAM_REDUCE_SIZE, Type, void>::type;
 
-        if (m_instance == nullptr) return value;
```

Comment:
Should we Kokkos Assert here or something instead then? (and elsewhere)

Reply:
The only place where `m_instance` is set, is in the constructor and we are using it directly after this line. Hence, I don't see much value in adding an assert here (but we could).

Reply:
ah got it, I'm fine with not checking if it is part of the class invariant that it is not null and not an individual precondition for one of the member functions

Reply:
Ah, but lets at least check it in the constructor then https://github.com/kokkos/kokkos/pull/6969/files#diff-3bf5843decd4b71c16ceb24b342b912271c46008d99ccc601d8f5b395d4fbb74L348